lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,70 @@
+require 'util/mount/miq_generic_mount_session'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    # NOTE: This line to be removed once manageiq-ui-class region change implemented.
+    @settings[:region] = "us-east-1" if @settings[:region].nil?
+    raise "username, password, and region are required values!" if @settings[:username].nil? || @settings[:password].nil? || @settings[:region].nil?
+    @host       = URI(@settings[:uri]).host
+    @mount_path = URI(@settings[:uri]).path
+  end
+
+  def connect
```

Comment:
Shouldn't we just not define this method and let it be inherited from the base class if we're just calling `super`?

Reply:
Here's your comment from 11 days ago (below) - "Why even bother having content here? I think it would be more clear to set @host and @mount_path on initialize and leave this method with no content with a comment that explains that you don't really have to "connect" to the object store."

Reply:
Hm, past me has a point ... can we remove the `super` call then? If we don't need it at all why call into the base class?

Reply:
Why do we need `super`? All `MiqGenericMountSession#connect` does is set `@mount_point` which, like you (and I) said, you're doing in `initialize`.

It sets it directly [here](https://github.com/ManageIQ/manageiq-gems-pending/blob/master/lib/gems/pending/util/mount/miq_generic_mount_session.rb#L129) then in a separate method [here](https://github.com/ManageIQ/manageiq-gems-pending/blob/master/lib/gems/pending/util/mount/miq_generic_mount_session.rb#L78) wouldn't that overwrite what you're doing in `initialize`?

Reply:
Ah, I see. Different variables (`mount_path` vs `mnt_point`). Well if this works then a method with just super in it is the same as the method not being present at all so we can remove it I guess.

Reply:
I still think this method should be removed. Do you think there's any reason to have a method with just `super` as the content @jerryk55?

Reply:
No other than your original comment.  I'll take it out...